File: OZ.API/OZ.Models/Addict.cs
using Microsoft.AspNetCore.Identity;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace OZ.Models
{
    public class Addict //: IdentityUser
    {
        [Key]
        public Guid OID { get; set; }
        public string AddictCode { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string FullName { get; set; }
        //public string FullName { get { return LastName + " " + FistName; } }
        public string OtherName { get; set; }
        public int GenderID { get; set; }
        public Guid? PlaceOfBirthID { get; set; }
        public DateTime? DateOfBirth { get; set; }
        public int? YearOfBirth { get; set; }
        public int? MonthOfBirth { get; set; }
        public int? DayOfBirth { get; set; }
        public string PemanentAddress { get; set; }
        public string CurrentAddress { get; set; }
        public string Profession { get; set; }
        public string PhoneNumber { get; set; }
        public string SocialNetworkAccount { get; set; }
        public Guid? EducationLevelID { get; set; }
        public string CitizenID { get; set; }
        public string CriminalConviction { get; set; }
        public string CriminalRecord { get; set; }
        public bool? Detoxed { get; set; }
        public string FartherName { get; set; }
        public string MotherName { get; set; }
        public string PartnerName { get; set; }
        public string Characteristics { get; set; }

        //public Guid JailID { get; set; }
        //public Guid RehabCampID { get; set; }
        public string Remarks1 { get; set; }
        public string Remarks2 { get; set; }
        public int? ManageType { get; set; }
        public bool? Complete { get; set; }
        public DateTime? CompleteDate { get; set; }
        public DateTime? CreateDate { get; set; }
        public Guid? CreateUser { get; set; }
        public string ImgLink { get; set; }
        public Guid? ManagePlaceID { get; set; } // de biet thuoc phuong nao quan ly

    }

    public class AddictBaseDto 
    {        
        public Guid OID { get; set; }
        public string AddictCode { get; set; }
        public string FullName { get; set; }
    }

    public class Addict2Dto
    {
        public Guid OID { get; set; }
        public string AddictCode { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int GenderID { get; set; }
        public Guid? PlaceOfBirthID { get; set; }
        public DateTime? DateOfBirth { get; set; }       
        public string PemanentAddress { get; set; }
        public string CurrentAddress { get; set; }
        public string Profession { get; set; }
        public string PhoneNumber { get; set; }
        public string SocialNetworkAccount { get; set; }
        public Guid? EducationLevelID { get; set; }
        public string CitizenID { get; set; }
        public string CriminalConviction { get; set; }
        public string CriminalRecord { get; set; }       
        public string FartherName { get; set; }
        public string MotherName { get; set; }
        public string PartnerName { get; set; }
        public string Characteristics { get; set; }
        public string ImgLink { get; set; }
        public double CorrectRatio { get; set; }
    }
}

File: Angular/src/app/Shared/Models/Addict.ts
//import * as internal from "stream";
import { AddictDrugs } from "../Models/AddictDrugs";
import { AddictManagePlace } from "../Models/AddictManagePlace";
import { AddictClassify } from "./AddictClassify";

export interface Addict {
    oid	: string;
    addictCode	: string;
    firstName	: string;
    lastName	: string;
    fullName	: string;
    otherName	: string;
    genderID	: number;
    placeOfBirthID	: string;
    dateOfBirth : Date;
    // yearOfBirth	:number;
    // monthOfBirth	:number;
    // dayOfBirth	: number;
    pemanentAddress	: string;
    currentAddress	: string;
    profession	: string;
    phoneNumber	: string;
    socialNetworkAccount	: string;
    educationLevelID	: string;
    citizenID	: string;
    criminalConviction	: string;
    criminalRecord	: string;
    detoxed	: boolean;
    fartherName	: string;
    motherName	: string;
    partnerName	: string;
    characteristics	: string;
    remarks1	: string;
    remarks2	: string;
    manageType : number;
    complete	:boolean;
    completeDate	: Date;
    createDate	:Date;
    createUser	: string;
    imgLink	: string;
    updCount : number;
    managePlaceID	: string;
    correctRatio: number;
    drugs : AddictDrugs[];
    classifys : AddictClassify[];
    places : AddictManagePlace[];
}

// export interface AddictBase {
//     oid	: string;
//     addictCode	: string;
//     fullName	: string;
// }